SCMUlate/apps/luDecomp/Codelets/LuDecomp.cpp

```diff
@@ -0,0 +1,222 @@
+#include "LuDecomp.hpp"
+#include <iomanip>
+
+// BENCH is externed so now we can access it directly instead of using a base register
+
+// sizes of codelets? big questions here
+// col -- read/write
+// diag -- read
+IMPLEMENT_CODELET(fwd_2048L,
+  float *col = this->getParams().getParamValueAs<float*>(1);
+  float *diag = this->getParams().getParamValueAs<float*>(2);
+
+  int i;
+  int j;
+  int k;
+  for (j=0; j<bots_arg_size_1; j++)
+    for (k=0; k<bots_arg_size_1; k++) 
+        for (i=k+1; i<bots_arg_size_1; i++)
+          col[i*bots_arg_size_1+j] = col[i*bots_arg_size_1+j] - diag[i*bots_arg_size_1+k]*col[k*bots_arg_size_1+j];
+);
+
+IMPLEMENT_CODELET(bdiv_2048L,
+  float *row = this->getParams().getParamValueAs<float*>(1);
+  float *diag = this->getParams().getParamValueAs<float*>(2);
+
+  int i; 
+  int j;
+  int k;
+  for (i=0; i<bots_arg_size_1; i++)
+    for (k=0; k<bots_arg_size_1; k++)
+    {
+      row[i*bots_arg_size_1+k] = row[i*bots_arg_size_1+k] / diag[k*bots_arg_size_1+k];
+      for (j=k+1; j<bots_arg_size_1; j++)
+        row[i*bots_arg_size_1+j] = row[i*bots_arg_size_1+j] - row[i*bots_arg_size_1+k]*diag[k*bots_arg_size_1+j];
+    }
+);
+
+IMPLEMENT_CODELET(bmod_2048L,
+  float *inner = this->getParams().getParamValueAs<float*>(1);
+  float *row = this->getParams().getParamValueAs<float*>(2);
+  float *col = this->getParams().getParamValueAs<float*>(3);
+
+  int i; 
+  int j;
+  int k;
+  for (i=0; i<bots_arg_size_1; i++)
+    for (j=0; j<bots_arg_size_1; j++)
+      for (k=0; k<bots_arg_size_1; k++)
+        inner[i*bots_arg_size_1+j] = inner[i*bots_arg_size_1+j] - row[i*bots_arg_size_1+k]*col[k*bots_arg_size_1+j];
+);
+
+IMPLEMENT_CODELET(lu0_2048L,
+  float *diag = this->getParams().getParamValueAs<float*>(1);
+  //printf("diag is %p\n", diag);
```

Comment:
It is best to remove these debugging code for the final version. Or use `SCMULATE_INFOMSG()`